community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/IndexPopulationJob.java

```diff
@@ -19,186 +19,106 @@
+ * {@link #addPopulator(IndexPopulator, IndexDescriptor, IndexConfiguration, org.neo4j.kernel.api.index.SchemaIndexProvider.Descriptor, String, FlippableIndexProxy, FailedIndexProxyFactory)}
+ * before {@link #run() running} this job.
  */
 public class IndexPopulationJob implements Runnable
 {
-    // NOTE: unbounded queue expected here
-    private final Queue<NodePropertyUpdate> queue = new ConcurrentLinkedQueue<>();
-    private final IndexDescriptor descriptor;
-
-    private final IndexConfiguration config;
-    private final SchemaIndexProvider.Descriptor providerDescriptor;
-
     private final IndexingService.Monitor monitor;
-    private final IndexPopulator populator;
-    private final FlippableIndexProxy flipper;
+    private final MultipleIndexPopulator populator;
     private final IndexStoreView storeView;
-    private final String indexUserDescription;
-    private final FailedIndexProxyFactory failureDelegate;
-    private final LogProvider logProvider;
-    private final Log log;
-    private final IndexCountsRemover indexCountsRemover;
     private final CountDownLatch doneSignal = new CountDownLatch( 1 );
     private final Runnable schemaStateChangeCallback;
 
     private volatile StoreScan<IndexPopulationFailedKernelException> storeScan;
     private volatile boolean cancelled;
 
-    public IndexPopulationJob( IndexDescriptor descriptor,
-                               IndexConfiguration config,
-                               SchemaIndexProvider.Descriptor providerDescriptor,
-                               String indexUserDescription,
-                               FailedIndexProxyFactory failureDelegateFactory,
-                               IndexPopulator populator,
-                               FlippableIndexProxy flipper,
-                               IndexStoreView storeView,
+    public IndexPopulationJob( IndexStoreView storeView,
                                LogProvider logProvider,
                                IndexingService.Monitor monitor,
                                Runnable schemaStateChangeCallback )
     {
-        this.descriptor = descriptor;
-        this.config = config;
-        this.providerDescriptor = providerDescriptor;
-        this.populator = populator;
-        this.flipper = flipper;
+        this.populator = new MultipleIndexPopulator( storeView, logProvider );
         this.storeView = storeView;
         this.schemaStateChangeCallback = schemaStateChangeCallback;
-        this.indexUserDescription = indexUserDescription;
-        this.failureDelegate = failureDelegateFactory;
-        this.logProvider = logProvider;
         this.monitor = monitor;
-        this.log = logProvider.getLog( getClass() );
-        this.indexCountsRemover = IndexCountsRemover.Factory.create( storeView, descriptor );
     }
 
+    /**
+     * Adds an {@link IndexPopulator} to be populated in this store scan. All participating populators must
+     * be added before calling {@link #run()}.
```

Comment:
Would be nice to not impose such complex contract on the users of this class. This could be done by passing list of populators to this job. Is this easy to do?
Otherwise it would be good to check this contract and throw if it is violated and have a test.